plugins/inputs/irqstat/irqstat.go

```diff
@@ -0,0 +1,128 @@
+package irqstat
+
+import (
+	"bufio"
+	"fmt"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"io/ioutil"
+	"log"
+	"strconv"
+	"strings"
+)
+
+type Irqstat struct {
+	Include []string
+}
+
+type IRQ struct {
+	ID     string
+	Fields map[string]interface{}
+	Tags   map[string]string
+}
```

Comment:
This type should use the domain appropriate names for an IRQ (device, type, total) or it should be removed and Accumulator should be passed into the parseInterrupts function.  You might think this will make testing difficult, but you can use testutil.Accumulator for that.

Side note: this type is essentially an internal type called Metric, but unfortunately Accumulator doesn't accept it.